include/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.hpp

```diff
@@ -0,0 +1,133 @@
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State/NumericalSolver.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace flight
+{
+
+using ostk::core::ctnr::Array;
+using ostk::core::ctnr::Pair;
+using ostk::core::filesystem::File;
+using ostk::core::types::Shared;
+using ostk::core::types::Size;
+using ostk::core::types::String;
+using ostk::core::types::Real;
+
+using ostk::math::object::MatrixXd;
+using ostk::math::object::VectorXd;
+using ostk::math::object::Vector3d;
+using ostk::math::curvefitting::Interpolator;
+using ostk::math::curvefitting::interpolator::BarycentricRational;
+using ostk::math::curvefitting::interpolator::CubicSpline;
+using ostk::math::curvefitting::interpolator::Linear;
+
+using ostk::physics::coord::Frame;
+using ostk::physics::data::Scalar;
+using ostk::physics::time::Instant;
+using ostk::physics::time::Duration;
+using ostk::physics::units::Mass;
+
+using TabulatedDynamics = ostk::astro::dynamics::Tabulated;
+using ostk::astro::trajectory::state::CoordinatesSubset;
+using ostk::astro::trajectory::state::NumericalSolver;
+
+#define DEFAULT_INTERPOLATION_TYPE Tabulated::InterpolationType::BarycentricRational
+
+/// @brief Stores information and metadata about a maneuver as performed by a spacecraft.
+class Maneuver
+{
+   public:
+    static const Shared<const Frame> AccelerationFrameSPtr;
+
+    enum class Type
+    {
+        ConstantThrust,
+        ConstantThrustAndDirection,
+        Custom
+    };
+
+    enum class InterpolationType
+    {
+
+        Linear,
+        BarycentricRational,
+        CubicSpline
+
+    };
+
+    Maneuver(
+        const Maneuver::Type& aManeuverType,
+        const Array<Instant>& anInstantArray,
+        const Array<Vector3d>& anAccelerationProfile,
+        const Shared<const Frame>& aFrameSPtr,
+        const Real& aMassFlowRate,
+        const InterpolationType& anInterpolationType = DEFAULT_INTERPOLATION_TYPE
+    );
+
+    bool isDefined();
+
+    Maneuver::Type getType();
+    Pair<Array<Instant>, Array<Vector3d>> getInstantsAndAccelerationProfile();
+    Real getMassFlowRate();
+    Instant getStartInstant();
+    Instant getEndInstant();
+    Duration getDuration();
+
+    Real estimateDeltaV();
+    Mass estimateDeltaMass();
+
+    Vector3d estimateThrustAccelerationAt(
+        const Instant& anInstant,
+        const Shared<const Frame>& aFrameSPtr,
+        const TabulatedDynamics::InterpolationType& anInterpolationType = DEFAULT_INTERPOLATION_TYPE
+    );
+
+    Shared<TabulatedDynamics> toTabulatedDynamics(
+        const Shared<const Frame>& aFrameSPtr,
+        const TabulatedDynamics::InterpolationType& anInterpolationType = DEFAULT_INTERPOLATION_TYPE
+    );
+
+    static Tabulated Load(const Maneuver::Type& aManeuverType, const File& aFile);
+
+   private:
+    const Type type_;
+    const Array<Instant> instants_;
+    const Array<Vector3d>& accelerationProfile_;
+    const Real massFlowRate_;
+    Shared<Interpolator> interpolatorSPtr_;
```

Comment:
- I think by having a constant mass flow rate we are unnecessarily limiting ourselves.
We can always have 2 constructors, that accept a Real, or a VectorXd, and then it's patched together under the hood.
- I don't see the value in having it's own interpolator...
- Agreed we should always store in an inertial frame like GCRF

Reply:
1. Yeah that's true, i'll change it back to an Array of either Reals or a VectorXd depending on whether I think its worth it to have ISP as well as Mass being specified
2. The reason I wanted to have its own interpolator is that I don't feel like the interpolator is going to change much, so I don't want to have to specify it as an argument everytime I want a quantity from the maneuver interpolated. I have as a memvar for the same reason that TabulatedDynamics has it as a memvar. Do you think it would be better just to leave the "best" interpolator for maneuvers (that we find from testing) as a default argument for any function requiring interpolating the internal quantities of the object? 
@vishwa2710

Reply:
Sorry just left a comment on why flowmassrate was a real
I see it's being agreed upon
so +1 on what @vishwa2710 has mentioned

Going back to the interpolator question or in general, they 'memvar' point that you bring up.

If we leave implementation aside, just from a "design" perspective, if a maneuver is a historical set of time-based numbers, it should be just that.

Any byproduct of that (for instance, a tabulated dynamics) could be created with many different "flavours" (the interpolator type being one of them for now). Having these "biproducts" as memvars "forces" them to be frozen with one "flavour" or another, whereas the "beauty" (imo) of the Maneuver object is that you store only what you need to recreate said bi-products. 

I see no issue in having default argument values for these "flavour options".

Reply:
Okay agreed, and I guess `TabulatedDynamics` is slightly different cause from a software perspective, it behaves like any other dynamics, and "needs" to have its own interpolator in order to produce computedContributions at any requested instant. But since the Maneuver object is free on its own and doesn't have to conform to the function signature of the `computeContributions` function from a parent `Dynamics` class, it makes more sense to keep it lighter-weight without its own interpolator that can be specified as an argument. That is my understanding.